waltz-test/src/main/python/waltz_ducktape/tests/validation/connection_interruption_test.py

```diff
@@ -0,0 +1,81 @@
+from waltz_ducktape.tests.produce_consume_validate import ProduceConsumeValidateTest
+from ducktape.mark.resource import cluster
+from ducktape.mark import parametrize
+from ducktape.cluster.cluster_spec import ClusterSpec
+from ducktape.utils.util import wait_until
+from time import sleep
+from random import randrange
+
+
+class ConnectionInterruptionTest(ProduceConsumeValidateTest):
+    """
+        Class for tests simulating network issues corresponding to Waltz Server and Waltz Storage nodes
+    """
+
+    MIN_CLUSTER_SPEC = ClusterSpec.from_list([
+        {'cpu': 1, 'mem': '1GB', 'disk': '25GB', 'additional_disks': {'/dev/sdb': '100GB'}, 'num_nodes': 3},
+        {'cpu': 1, 'mem': '3GB', 'disk': '15GB', 'num_nodes': 2},
+        {'cpu': 1, 'mem': '1GB', 'disk': '25GB', 'num_nodes': 1}])
+
+    def __init__(self, test_context):
+        super(ConnectionInterruptionTest, self).__init__(test_context=test_context)
+
+    @cluster(cluster_spec=MIN_CLUSTER_SPEC)
+    @parametrize(num_active_partitions=1, txn_per_client=100, num_clients=2, interval=600, timeout=360,
+                 interruption_duration=10, num_interruptions=3, delay_between_interruptions=5)
+    @parametrize(num_active_partitions=4, txn_per_client=100, num_clients=2, interval=1000, timeout=300,
+                 interruption_duration=20, num_interruptions=1, delay_between_interruptions=20)
+    def test_produce_consume_with_network_interruption(self, num_active_partitions, txn_per_client, num_clients, interval, timeout,
+                                                interruption_duration, num_interruptions, delay_between_interruptions):
+        validation_cmd = self.client_cli.validate_txn_cmd(num_active_partitions, txn_per_client, num_clients, interval)
+        self.run_produce_consume_validate(lambda: self.produce_consume_with_network_interruption(validation_cmd, timeout, interruption_duration, num_interruptions, delay_between_interruptions, num_active_partitions, interval))
```

Comment:
NIT: line is too long. Let's split it into multiple lines.
Same applies for other lines as well.